xc/device.ml

```diff
@@ -1414,8 +1426,72 @@ module PCI = struct
       Forkhelpers.execute_command_get_output !Resources.rmmod ["i915"] in ()
 
   let procfs_nvidia = "/proc/driver/nvidia/gpus"
+  let nvidia_smi    = "/usr/bin/nvidia-smi"
+  let nvidia_manage = "/usr/lib/nvidia/sriov-manage"
 
-  let nvidia_smi = "/usr/bin/nvidia-smi"
+  (** [num_vfs devstr] returns the number of PCI VFs of [devstr] or 0 if
+    * [devstr] is not an SRIOV device
+    *)
+  let num_vfs devstr =
+    let path = Printf.sprintf "/sys/bus/pci/devices/%s/sriov_numvfs" devstr in
+    try
+      Some (Unixext.string_of_file path |> String.trim |> int_of_string)
```

Comment:
Is `Some 0` different from `None`, i.e. could the return value here be an integer, with default 0 if there are no VFs?

Reply:
Could return an integer - this is left over from an earlier design where there was a distinction.